### DISCUSSION TITLE
Additional FuncMap for templates

### DISCUSSION THREAD

**Discussion by Sinderella** [aquasecurity/trivy#4940] [Ideas]
### Description

I am pretty new to both go and go template.

I am trying to write my own output template for MultiMarkdown table (without HTML tags) format. I see that someone has already implemented a markdown compatible format and it's been rejected (https://github.com/aquasecurity/trivy/pull/3202#issuecomment-1500497668). However, I don't wish to use HTML. The issue I am experiencing now is that there are a few useful functions that I'd like to have. I am not sure if I can do these without asking for the new FuncMap to be added or if I can extend those myself.

- Find/Replace function: As some vulnerability's description contains newline characters, it breaks markdown table. I need to get rid of it or have it replaced with `\newline` for post-processing. It might be useful in other cases as well, for cleanups.
- Title case function: I couldn't find a way to do a title case on some field such as `Severity`. I can do post-processing for keywords, but still, not ideal.

I am more than happy to create a PR.

### Target

None

### Scanner

None

**Comment by knqyf263** [comment]
You can use these functions. It should help.
http://masterminds.github.io/sprig/strings.html

**Reply by Sinderella** [reply]
Thank you! I was trying to use `title` but it didn't work, I thought the app doesn't support those functions.

I just found out that `title` doesn't work on the all capitalised words like the severity.

```
| {{ title .Severity }} {{ title "test" }} {{ title "TEST" }} {{ lower "TEST" | title }} |
```

gives me the following output

```
| HIGH Test TEST Test |
```

So the functions work after all, just that one didn't work as expected. Cheers!
